integrations/nuclei_analyzer/app.py

```diff
@@ -0,0 +1,240 @@
+            url,
+            "-jsonl",
+        ]
+
+        if template_dirs:
+            for template_dir in template_dirs:
+                if template_dir not in temp_dirs:
+                    return False, {
+                        "success": False,
+                        "error": "Invalid template directory",
+                        "details": f"Invalid template directory: {template_dir}",
+                    }
+                else:
+                    command.extend(["-t", template_dir])
+
+        logger.info(f"Running command: {' '.join(command)}")
+
+        report = subprocess.run(
+            command,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            timeout=1200,
+            check=True,
+        )
+
+        if report.returncode != 0:
+            error_msg = report.stderr.strip()
+            return False, {
+                "success": False,
+                "error": "Failed to run Nuclei",
+                "details": error_msg,
+            }
+
+        output_lines = [
+            line.strip() for line in report.stdout.split("\n") if line.strip()
+        ]
+        parsed_reports = []
+
+        for line in output_lines:
+            try:
+                parsed_reports.append(json.loads(line))
+            except json.JSONDecodeError as e:
+                logger.warning(f"Failed to parse JSON line: {line}, error: {str(e)}")
+                continue
+
+        return True, {
+            "success": True,
+            "result": parsed_reports,
+            "scan_status": "COMPLETED",
+            "statistics": {"total_findings": len(parsed_reports)},
+        }
+
+    except subprocess.TimeoutExpired:
+        return False, {
+            "success": False,
+            "error": "Scan timed out",
+            "scan_status": "TIMEOUT",
+        }
+    except Exception as e:
+        return False, {
+            "success": False,
+            "error": "An unexpected error occurred",
+            "details": str(e),
+            "scan_status": "ERROR",
+        }
+
+
+def async_nuclei_scan(task_id: str, url: str, template_dirs: list = None):
+    try:
+        _, report = run_nuclei_command(url, template_dirs)
+
+        task_store[task_id].update(
+            {
+                "status": "completed",
+                "report": report,
+                "completed_at": datetime.utcnow().isoformat(),
+            }
+        )
+        logger.info(f"Task {task_id} reports stored successfully")
+    except Exception as e:
+        task_store[task_id].update(
+            {
+                "status": "error",
+                "error": str(e),
+                "completed_at": datetime.utcnow().isoformat(),
+            }
+        )
+
+
+@app.route("/run-nuclei", methods=["GET"])
+def get_nuclei_scan_status():
+    task_id = request.args.get("key")
+    if not task_id:
+        return jsonify({"success": False, "error": "Missing task key parameter"}), 400
```

Comment:
jsonify function is not needed anywhere when you return a list or json. The framework handles it https://flask.palletsprojects.com/en/stable/quickstart/#about-responses